src/date-picker/src/panel/date.tsx

```diff
@@ -19,6 +19,14 @@ export default defineComponent({
   name: 'DatePanel',
   props: useCalendar.props,
   setup (props) {
+    onMounted(() => {
+      if (props.actions?.includes('confirm')) {
+        warn(
+          'DatePanel',
+          'The `confirm` action is not supported for n-date-picker of `date` type'
+        )
+      }
+    })
```

Comment:
```suggestion
if (__DEV__) {
    watchEffect(() => {
      if (props.actions?.includes('confirm')) {
        warn(
          'date-picker',
          'The `confirm` action is not supported for n-date-picker of `date` type'
        )
      }
    })
  }
```

不要打进生产，然后用 `watchEffect` 和 `warnOnce`